### PR TITLE
fix: error when unlinking google account

### LINKED ISSUE #14720: [CAL-3519] Error when trying to unlink a google account
I just stumbled into an error in prod when trying to unlink a google account.

IMO we should handle unlinking in a separate handler besides updateProfile.

image

Here's the redacted stack trace ([View in Sentry](https://calcom.sentry.io/issues/5244526249/?project=4504055091625984&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=14d&stream_index=4)):

```
 Error:  {"name":"PrismaClientValidationError","clientVersion":"5.4.2"}
Something went wrong Error: An error occured while querying the database.
    at redactError (/var/task/apps/web/.next/server/chunks/30267.js:1:621)
    ... 8 lines matching cause stack trace ...
    at async /var/task/node_modules/@sentry/nextjs/cjs/common/wrapApiHandlerWithSentry.js:153:35 {
  code: 'INTERNAL_SERVER_ERROR',
  name: 'TRPCError',
  [cause]: Error: An error occured while querying the database.
      at redactError (/var/task/apps/web/.next/server/chunks/30267.js:1:621)
      at /var/task/apps/web/.next/server/chunks/46279.js:1:1628
      at async callRecursive (file:///var/task/node_modules/@trpc/server/dist/unstable-core-do-not-import/procedureBuilder.mjs:125:32)
      at async procedure (file:///var/task/node_modules/@trpc/server/dist/unstable-core-do-not-import/procedureBuilder.mjs:155:24)
      at async inputToProcedureCall (file:///var/task/node_modules/@trpc/server/dist/unstable-core-do-not-import/http/resolveHTTPResponse.mjs:47:22)
      at async Promise.all (index 0)
      at async resolveHTTPResponse (file:///var/task/node_modules/@trpc/server/dist/unstable-core-do-not-import/http/resolveHTTPResponse.mjs:184:37)
      at async file:///var/task/node_modules/@trpc/server/dist/adapters/node-http/nodeHTTPRequestHandler.mjs:82:9
      at async file:///var/task/node_modules/@trpc/server/dist/adapters/next.mjs:47:9
      at async /var/task/node_modules/@sentry/nextjs/cjs/common/wrapApiHandlerWithSentry.js:153:35
}
```

\_Originally posted by @zomars in [https://github.com/calcom/cal.com/issues/14082#issuecomment-2072995264\_](https://github.com/calcom/cal.com/issues/14082#issuecomment-2072995264\_)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3519](https://linear.app/calcom/issue/CAL-3519/error-when-trying-to-unlink-a-google-account)</sub>

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Fixes #14720
- Fixes CAL-3519
- Removed unlinking logic from updateProfile handler
- Made new endpoint specific for unlinking accounts

https://github.com/calcom/cal.com/assets/3504472/b1b4623e-fe1f-496a-97b3-f8b4f5376c40

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

```bash
yarn test unlinkConnectedAccount
```

- Are there environment variables that should be set?
  - No
- What are the minimal test data to have?
  - Having identityProvider and identityProviderId
- What is expected (happy path) to have (input and output)?
  - Have an account with email and password
  - Link a google account with the same email
  - Go to profile settings and unlink it
- Any other important info that could help to test that PR
  - Nope

### CODE REVIEW

File: apps/web/pages/settings/my-account/profile.tsx [hunk 1]
```diff
@@ -439,10 +448,7 @@ const ProfileView = () => {
             <Button
               color="primary"
               onClick={() => {
-                updateProfileMutation.mutate({
```
Comment (zomars) [inline apps/web/pages/settings/my-account/profile.tsx:442]: It didn't make much sense to have this being part of updateProfile. updateProfile is doing too much already.

File: packages/trpc/server/routers/loggedInViewer/unlinkConnectedAccount.handler.spec.ts
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/unlinkConnectedAccount.handler.spec.ts:1]: Added simple unit/integration test

File: packages/trpc/server/routers/loggedInViewer/updateProfile.handler.ts [hunk 2]
```diff
@@ -74,9 +73,6 @@ export const updateProfileHandler = async ({ ctx, input }: UpdateProfileOptions)
   const secondaryEmails = input?.secondaryEmails || [];
   delete input.secondaryEmails;
 
-  const unlinkConnectedAccount = input?.unlinkConnectedAccount || false;
-  delete input.unlinkConnectedAccount;
```
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/updateProfile.handler.ts:78]: TBH. This feels like a hack. Separated to new handler.

File: packages/trpc/server/routers/loggedInViewer/unlinkConnectedAccount.handler.ts
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/unlinkConnectedAccount.handler.ts:1]: Single purpose handler in here

**Review by zomars — COMMENTED** [review]
Self review done

**Review by Udit-takkar — APPROVED** [review]
LGTM. Tested and worked fine

**Pull request merged**